File: web3rpc/rpc-specs/code-samples/python/debug/profiling/setBlockProfileRate.py
import json
from web3 import Web3
from web3py_ext import extend

host = "https://public-en-kairos.node.kaia.io"

rate = 4

w3 = Web3(Web3.HTTPProvider(host))
debug_response = w3.debug.set_block_profile_rate(rate)

print(json.loads(debug_response.response.data))

File: web3rpc/rpc-specs/code-samples/javascript/debug/profilling/setBlockProfileRate.js
const { JsonRpcProvider } = require("@kaiachain/ethers-ext");
(() => {
  const provider = new JsonRpcProvider("https://public-en-kairos.node.kaia.io");
  const rate = 3;

  provider.debug
    .setBlockProfileRate(rate, {}, (err, data, response) => {})
    .then((data) => {
      console.log(data);
    });
})();
